src/core/src/types/arrays.rs

```diff
@@ -73,28 +59,13 @@ impl<T: Zero + Clone> Array2<T> {
     /// a.resize_if_different((8, 9));
     /// assert_eq!(a[(3, 3)], 0.0);
     /// ```
-    pub fn resize_if_different(&mut self, size: (Ix, Ix)) {
-        if self.0.shape() != &[size.0, size.1] {
+    pub fn resize_if_different(&mut self, size: (usize, usize)) {
```

Comment:
This is not really a comment on this PR, but this behavior is not what is expected given the name of the function. I do not expect a `resize` operation to overwrite values that are contained in the new size, I expect something in the line of a [`std::vector.resize()`](http://en.cppreference.com/w/cpp/container/vector/resize) in C++